tests/driver/test_pymysql_driver.py

```diff
@@ -0,0 +1,115 @@
+        self.mock_pymysql_connection = MockPyMySQLConnection(cursor=mock_cursor)
+
+     def test_mysql_connection_ssl_enable(self):
+         """Test creating mysql connection with ssl enabled"""
+         
+         mysql_connection_params = {
+            'connectTimeout': '30', 
+            'dbname': '', 
+            'groupId': 'C777F06B-202E-4480-B475-FA416154D458', 
+            'host': 'mysql-test.mysql.database.azure.com', 
+            'password': 'samplePass123', 
+            'user': 'sampleUser',
+            'ssl': 'require',
+            'ssl.ca': 'path_to_ssl_cert'
+            }
+         
+         with mock.patch('pymysql.connect', new=mock.Mock(return_value=self.mock_pymysql_connection)):
+            mysqlConnection = MySQLConnection(mysql_connection_params)
+         
+         expected_connection_options = {
+             'connect_timeout': 30, 
+             'database': '', 
+             'host': 'mysql-test.mysql.database.azure.com', 
+             'password': 'samplePass123', 
+             'port': 3306, 
+             'user': 'sampleUser',
+             'ssl':{
+                 'ca':'path_to_ssl_cert'
+             }
+             }   
+         print(mysqlConnection.connection_options)
+         self.assertDictEqual(mysqlConnection.connection_options, expected_connection_options)
+         self.assertIs(mysqlConnection.connection, self.mock_pymysql_connection)
+         self.assertTrue(mysqlConnection.autocommit)
+         self.assertEqual(mysqlConnection.default_database, constants.DEFAULT_DB[constants.MYSQL_PROVIDER_NAME])
+         
+     def test_mysql_connection_ssl_disable(self):
+         """Test creating mysql connection with ssl disabled"""
+         
+         mysql_connection_params = {
+            'connectTimeout': '30', 
+            'dbname': '', 
+            'groupId': 'C777F06B-202E-4480-B475-FA416154D458', 
+            'host': 'mysql-test.mysql.database.azure.com', 
+            'password': 'samplePass123', 
+            'user': 'sampleUser'}
+         
+         with mock.patch('pymysql.connect', new=mock.Mock(return_value=self.mock_pymysql_connection)):
+            mysqlConnection = MySQLConnection(mysql_connection_params)
+         
+         expected_connection_options = {
+             'connect_timeout': 30, 
+             'database': '', 
+             'host': 'mysql-test.mysql.database.azure.com', 
+             'password': 'samplePass123', 
+             'port': 3306, 
+             'user': 'sampleUser'}   
+         print(mysqlConnection.connection_options)
+         self.assertDictEqual(mysqlConnection.connection_options, expected_connection_options)
+         self.assertIs(mysqlConnection.connection, self.mock_pymysql_connection)
+         self.assertTrue(mysqlConnection.autocommit)
+         self.assertEqual(mysqlConnection.default_database, constants.DEFAULT_DB[constants.MYSQL_PROVIDER_NAME])
+         
+     def test_mysql_connection_dbname_provided(self):
```

Comment:
Can add a testcase where port is not provided in connection_options.

Reply:
added